apps/web/src/services/dashboards/widgets/cost-widgets/aws-data-transfer-cost-trend/AWSDataTransferCostTrendWidget.vue

```diff
@@ -113,58 +111,76 @@ const { widgetState, widgetFrameProps, widgetFrameEventHandlers } = useWidget(pr
         });
         return sortBy(chartData, widgetState.dateRange, DATE_FIELD_NAME);
     }),
     tableFields: computed<Field[]>(() => {
-        const _textOptions: Field['textOptions'] = {
-            type: state.fieldsKey === 'cost_sum' ? 'cost' : 'number',
+        const _textOptions: Field['textOptions'] = state.fieldsKey === 'cost' ? {
+            type: 'cost',
+        } : {
+            type: 'usage',
+            unitPath: 'usage_unit',
         };
 
-        const refinedFields = getWidgetTableDateFields(widgetState.granularity, widgetState.dateRange, _textOptions, state.fieldsKey);
+        const refinedFields = getWidgetTableDateFields(widgetState.granularity, widgetState.dateRange, _textOptions, 'value_sum');
 
         // set width of table fields
         const groupByFieldWidth = refinedFields.length > 4 ? '28%' : '34%';
         const otherFieldWidth = refinedFields.length > 4 ? '18%' : '22%';
 
-        const fixedFields = [{
+        const groupByField: Field = {
             label: USAGE_TYPE_VALUE_KEY,
             name: USAGE_TYPE_VALUE_KEY,
             width: groupByFieldWidth,
-        }];
-        if (state.fieldsKey === 'usage_quantity_sum') {
-            fixedFields.push({
-                label: 'Usage Unit',
-                name: 'usage_unit',
-                width: otherFieldWidth,
-            });
-        }
+        };
         return [
-            ...fixedFields,
+            groupByField,
             ...refinedFields.map((field) => ({ ...field, width: otherFieldWidth })),
         ];
     }),
-    tableData: computed<WidgetTableData[]>(() => {
-        const tableData = sortArrayInObjectArray(
-            getRefinedDateTableData(state.data?.results, widgetState.dateRange, ['cost_sum', 'usage_quantity_sum']),
+    tableData: computed<Data[]>(() => {
+        const tableData: Data[] = sortArrayInObjectArray(
+            getRefinedDateTableData(state.data?.results, widgetState.dateRange, ['value_sum']),
             DATE_FIELD_NAME,
-            ['cost_sum', 'usage_quantity_sum'],
-        );
+            ['value_sum'],
+        ).map((data) => {
+            let value = data[USAGE_TYPE_VALUE_KEY] ?? 'Unknown';
+            if (state.fieldsKey === 'usage_quantity') value = `${value}${data.usage_unit ? ` (${data.usage_unit})` : ''}`;
+            return {
+                ...data,
+                [USAGE_TYPE_VALUE_KEY]: value,
+            };
+        });
+
         return tableData;
     }),
-    legends: computed<Legend[]>(() => getXYChartLegends(state.data?.results, USAGE_TYPE_VALUE_KEY)),
+    legends: computed<Legend[]>(() => {
+        const data = state.data?.results ?? [];
+        const legends: Legend[] = getXYChartLegends(data, USAGE_TYPE_VALUE_KEY)
+            .map((l, i) => {
+                let label = l.label;
+                if (state.fieldsKey === 'usage_quantity') label = `${label}${data[i]?.usage_unit ? ` (${data[i]?.usage_unit})` : ''}`;
+                return { ...l, label };
```

Comment:
note: note: the legend data must be refined as the same with the table data.
If that is empty, must be converted to 'Unknown'. -> `getXYChartLegends` helper does it.
And `usage_unit` must be displayed together.